### PR TITLE
feat(quarto): Add Quarto module

### PR DESCRIPTION
**Pull request by Armavica**
#### Description

I added a module for the [Quarto](https://quarto.org/) publishing system on the model of the Typst module.

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [x] I have updated the documentation accordingly.
- [x] I have updated the tests accordingly.

### CODE REVIEW

File: src/modules/quarto.rs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                "symbol" => Some(config.symbol),
+                _ => None,
+            })
+            .map_style(|variable| match variable {
+                "style" => Some(Ok(config.style)),
+                _ => None,
+            })
+            .map(|variable| match variable {
+                "version" => {
+                    let version = context.exec_cmd("quarto", &["--version"])?.stdout;
```
Comment (davidkna) [inline src/modules/quarto.rs:34]: Please handle the trailing newline and update the command mocks to add it.

File: docs/config/README.md [hunk 2]
```diff
@@ -3601,6 +3602,39 @@ detect_extensions = []
...
+- The current directory contains a `_quarto.yml` file
+- The current directory contains any `*.qmd` file
+
+### Options
+
+| Option              | Default                              | Description                                                               |
+| ------------------- | ------------------------------------ | ------------------------------------------------------------------------- |
+| `format`            | `'via [$symbol($version )]($style)'` | The format for the module.                                                |
+| `version_format`    | `'v${raw}'`                          | The version format. Available vars are `raw`, `major`, `minor`, & `patch` |
+| `symbol`            | `'⨁ '`                               | A format string representing the symbol of Quarto                         |
```
Comment (jankatins) [inline docs/config/README.md:3620]: One space too many in the default column, at least on github

Comment (davidkna) [reply docs/config/README.md:3620]: This is what the formatter settled on, so it should be fine.

File: docs/config/README.md [hunk 3]
```diff
@@ -3601,6 +3602,39 @@ detect_extensions = []
...
+| `detect_files`      | `['_quarto.yml']`                    | Which filenames should trigger this module.                               |
+| `detect_folders`    | `[]`                                 | Which folders should trigger this module.                                 |
+| `disabled`          | `false`                              | Disables the `quarto` module.                                             |
+
+### Variables
+
+| Variable      | Example   | Description                                       |
+| ------------- | --------- | ------------------------------------------------- |
+| version       | `1.4.549` | The version of `quarto`, alias for quarto_version |
+| typst_version | `default` | The current Typst version                         |
```
Comment (jankatins) [inline docs/config/README.md:3632]: I don'T see anything parsing/filling that variable. Is this needed/wanted?

In any case: given that quarto bundles typst, what's the value in this, given that in most (or even all cases: can you overwrite the typst version by installing a newer version), the quarto version will determine the typst version?

File: src/modules/quarto.rs [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    #[test]
+    fn read_quarto_present() -> io::Result<()> {
+        let dir = tempfile::tempdir()?;
+
+        File::create(dir.path().join("test.qmd"))?.sync_all()?;
+
+        let actual = ModuleRenderer::new("quarto").path(dir.path()).collect();
+        let expected = Some(format!(
+            "via {}",
+            Color::Rgb(0, 147, 167).bold().paint("⨁ v1.4.549 ")
```
Comment (jankatins) [inline src/modules/quarto.rs:84]: Will this test pass if quarto is not installed during the CI run?

File: src/modules/quarto.rs (hunk 1 above)
Comment (Armavica) [reply src/modules/quarto.rs]: Thank you, done.

File: docs/config/README.md (hunk 3 above)
Comment (Armavica) [reply docs/config/README.md]: Indeed, that's a copy/paste fail. I removed the variable.

**Comment by jankatins** [comment]
The commit and PR message is still mentioning Typst

**Comment by Armavica** [comment]
> The commit and PR message is still mentioning Typst

It was to credit the Typst module that I copy/pasted and adapted for this PR. But I can remove this mention if you want.

**Comment by davidkna** [comment]
LGTM, but please fix the conflicts.

**Comment by Armavica** [comment]
@davidkna Done!

**Review by davidkna — APPROVED** [review]

**Comment by andytom** [comment]
Thank you for your contribution @Armavica and thank you for reviewing @davidkna.

**Pull request merged**